dask_cloudprovider/providers/azure/azureml.py

```diff
@@ -0,0 +1,751 @@
+        )
+
+        ### DATASTORES
+        self.datastores = datastores
+
+        ### FUTURE EXTENSIONS
+        self.kwargs = kwargs
+
+        ### RUNNING IN MATRIX OR LOCAL
+        self.same_vnet = None
+
+        ### GET RUNNING LOOP
+        self._loop_runner = LoopRunner(loop=None, asynchronous=asynchronous)
+        self.loop = self._loop_runner.loop
+
+        self.abs_path = pathlib.Path(__file__).parent.absolute()
+
+        ### INITIALIZE CLUSTER
+        super().__init__(asynchronous=asynchronous)
+
+        if not self.asynchronous:
+            self._loop_runner.start()
+            self.sync(self.get_defaults)
+            self.sync(self.create_cluster)
+
+    async def get_defaults(self):
+        self.config = dask.config.get("cloudprovider.azure", {})
+
+        if self.experiment_name is None:
+            self.experiment_name = self.config.get("experiment_name")
+
+        if self.initial_node_count is None:
+            self.initial_node_count = self.config.get("initial_node_count")
+
+        if self.jupyter is None:
+            self.jupyter = self.config.get("jupyter")
+
+        if self.jupyter_port is None:
+            self.jupyter_port = self.config.get("jupyter_port")
+
+        if self.dashboard_port is None:
+            self.dashboard_port = self.config.get("dashboard_port")
+
+        if self.scheduler_port is None:
+            self.scheduler_port = self.config.get("scheduler_port")
+
+        if self.additional_ports is None:
+            self.additional_ports = self.config.get("additional_ports")
+
+        if self.admin_username is None:
+            self.admin_username = self.config.get("admin_username")
+
+        if self.admin_ssh_key is None:
+            self.admin_ssh_key = self.config.get("admin_ssh_key")
+
+        if self.datastores is None:
+            self.datastores = self.config.get("datastores")
+
+        ### PARAMETERS TO START THE CLUSTER
+        self.scheduler_params = {}
+        self.worker_params = {}
+
+        ### scheduler and worker parameters
+        self.scheduler_params["--jupyter"] = True
+
+        if self.use_gpu:
+            self.scheduler_params["--use_gpu"] = True
+            self.scheduler_params["--n_gpus_per_node"] = self.n_gpus_per_node
+            self.worker_params["--use_gpu"] = True
+            self.worker_params["--n_gpus_per_node"] = self.n_gpus_per_node
+
+        ### CLUSTER PARAMS
+        self.max_nodes = self.compute_target.serialize()["properties"]["properties"][
+            "scaleSettings"
+        ]["maxNodeCount"]
+        self.scheduler_ip_port = None
+        self.workers_list = []
+        self.URLs = {}
+
+        ### SANITY CHECKS
+        ###-----> initial node count
+        if self.initial_node_count > self.max_nodes:
+            self.initial_node_count = self.max_nodes
+
+    def __print_message(self, msg, length=80, filler="#", pre_post=""):
+        print(f"{pre_post} {msg} {pre_post}".center(length, filler))
```

Comment:
I have started using `logging` and added the logs where deemed necessary. This said, AzureML service maintains its own logs so this will be somewhat additional. More to it, we still would like to communicate with the user about the different stages the `Experiment-Run` is going through so I kept the `self.__print_mesage(...)` method and added `logger` call there.